src/signal.cr

```diff
@@ -132,6 +137,28 @@ enum Signal : Int32
     Signal::PIPE.ignore
     Signal::CHLD.reset
   end
+
+  # :nodoc:
+  def self.setup_sigfault_handler
+    return unless @@setup_sigfault_handler.test_and_set
+
+    altstack = LibC::StackT.new
+    action = LibC::Sigaction.new
+
+    altstack.ss_sp = LibC.malloc(STKSZ)
+    altstack.ss_size = STKSZ
+    altstack.ss_flags = 0
+    LibC.sigaltstack(pointerof(altstack), Pointer(LibC::StackT).null)
+
+    sa_mask = uninitialized LibC::SigsetT
+    LibC.sigemptyset(pointerof(sa_mask))
+    action.sa_mask = sa_mask
```

Comment:
You should be able to directly use a pointer to the struct member. That would be a more literal translation of the C implementation.

```suggestion
    LibC.sigemptyset(pointerof(action.sa_mask))
```

Reply:
Unfortunately, that doesn't compile. Seems `pointerof` doesn't work with struct members like that.

Reply:
Sry, I missed the `@` for accessing an ivar: `pointerof(action.@sa_mask)` works.

Reply:
oh nice, that works!